packages/orm/src/EntityManager.ts

```diff
@@ -1639,14 +1640,16 @@ async function validateReactiveRules(
   const p1 = Object.values(todos).flatMap((todo) => {
     const entities = [...todo.inserts, ...todo.updates, ...todo.deletes];
     // Find each statically-declared reactive rule for the given entity type
-    const rules = getBaseAndSelfMetas(todo.metadata).flatMap((m) => m.config.__data.reactiveRules);
```

Comment:
This `todo.metadata` was always the base type, i.e. we would group Task+TaskOld+TaskNew all into the same `todo`, which is good to issue just 1 SQL call.

But then we were only checking the "base and self" meta's list of reactive rules to run.

Now that each reactive rule knows the subtype its for, we could probably just push all of the rules onto the base meta, and then filter out ones for subtypes...

An optimization for another PR someday